control-statistics/src/main/java/fi/nls/oskari/control/statistics/user/DeleteUserIndicatorHandler.java

```diff
@@ -16,23 +16,29 @@
 @OskariActionRoute("DeleteUserIndicator")
 public class DeleteUserIndicatorHandler extends ActionHandler {
     private static final Logger LOG = LogFactory.getLogger(GetUserIndicatorsHandler.class);
-    private UserIndicatorService userIndicatorService = new UserIndicatorServiceImpl();
+    private StatisticalIndicatorService indicatorService;
+
+    @Override
+    public void init() {
+        super.init();
+        if (indicatorService == null) {
+            indicatorService = OskariComponentManager.getComponentOfType(StatisticalIndicatorService.class);
+        }
+    }
 
     public void handleAction(ActionParameters params) throws ActionException {
         // user indicators are user content so deleting one requires to be logged in
         params.requireLoggedInUser();
 
-        int id  = params.getRequiredParamInt(ActionConstants.PARAM_ID);
-        UserIndicator ui = userIndicatorService.find(id);
-        if(ui == null) {
-            throw new ActionParamsException("Unknown indicator");
+        int id = params.getRequiredParamInt(ActionConstants.PARAM_ID);
+        if (!indicatorService.delete(id, params.getUser().getId())) {
+            throw new ActionParamsException("Unknown indicator/not the owner of the indicator.");
         }
-        if(params.getUser().getId() != ui.getUserId()) {
-            throw new ActionDeniedException("User has no right to delete indicator");
+        LOG.info("Deleting indicator", id);
+        if (!indicatorService.delete(id, params.getUser().getId())) {
```

Comment:
`indicatorService.delete(id, params.getUser().getId())` is being called twice.

Reply:
Good catch :+1: